src/commands/appdistribution-distribute.ts

```diff
@@ -0,0 +1,134 @@
+import * as fs from "fs-extra";
+import * as crypto from "crypto";
+
+import * as Command from "../command";
+import * as utils from "../utils";
+import * as requireAuth from "../requireAuth";
+import * as req from "../appdistribution/requests";
+import { FirebaseError } from "../error";
+import { Distribution, DistributionFileType } from "../appdistribution/distribution";
+
+function ensureFileExists(file: string, message = ""): void {
+  if (!fs.existsSync(file)) {
+    throw new FirebaseError(`File ${file} does not exist: ${message}`);
+  }
+}
+
+function getAppId(appId: string): string {
+  if (!appId) {
+    throw new FirebaseError("set the --app option to a valid Firebase app id and try again");
+  }
+  return appId;
+}
+
+function getDistribution(distributionFile: string): Distribution {
```

Comment:
Why not just move this logic to the constructor of `Distribution`? It seems like validating the constructor args would make more sense there.

Reply:
Agreed, moved!